File: core/src/com/alta_v2/physics/executionContext/reserveData/ReservableBoolean.java
package com.alta_v2.physics.executionContext.reserveData;

import com.alta_v2.physics.executionContext.Reservable;
import com.alta_v2.physics.executionContext.Tenant;
import lombok.Getter;

/**
 * Provides the reservable object for boolean value.
 */
public class ReservableBoolean extends Reservable<Boolean> {

    @Getter
    private long changeTime;

    /**
     * Initialize new instance of {@link ReservableBoolean}.
     */
    public ReservableBoolean() {
        this.value = false;
    }

    /**
     * Gets the reserved value.
     */
    public boolean getValue() {
        return this.value;
    }

    public ReservableBoolean setValue(boolean value, Tenant tenant) {
        super.setValue(value, tenant);
        this.changeTime = System.currentTimeMillis();
        return this;
    }
}

File: core/src/com/alta_v2/physics/executionContext/reserveData/ReservableBoolean.kt
package com.alta_v2.physics.executionContext.reserveData

import com.alta_v2.physics.executionContext.Reservable
import com.alta_v2.physics.executionContext.Tenant

/**
 * Provides the reservable object for boolean value.
 */
class ReservableBoolean : Reservable<Boolean>() {

    private var changeTime: Long = 0

    init {
        this.value = false
    }

    override fun setValue(value: Boolean, tenant: Tenant): ReservableBoolean {
        super.setValue(value, tenant)
        changeTime = System.currentTimeMillis()
        return this
    }

    fun getChangeTime() = changeTime

    fun getValue() = value ?: false
}